hostmgr/cluster.go

```diff
@@ -149,7 +150,10 @@ func (c *Cluster) CreateNewHost(serial string) (*Host, error) {
 	newHost.Cabinet = cabinet
 	newHost.MachineOnCabinet = machineOnCabinet
 	newHost.MachineID = machineID
-	newHost.Hostname = machineID[:16]
+	if newHost.InternalAddr != nil {
```

Comment:
What if InternalAddr is nil?

Reply:
the thing is that this function is run few times, and in each run it  adds some extra info ( I am not sure why it works this ways, its kinda legacy, maybe we can later improve)